This dataset portrays an analysis between which NFL player has the most passing yards, rushing yards, rushing touchdowns, fumbles, and recieving touchdowns. The data analysis also portrays which age range of NFL players perform the best.

In [1]:
#importing code from https://github.com/bendominguez0111/fantasy-csv-data/blob/master/yearly/2021.csv
!pip install seaborn --upgrade
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import missingno as msno
import seaborn as sns
import os
import matplotlib.offsetbox as AnnotationBox

df = pd.read_csv("https://raw.githubusercontent.com/bendominguez0111/fantasy-csv-data/master/yearly/2021.csv")
df.head()

Unnamed: 0             Player   Tm Pos  Age   G  GS    Tgt    Rec  \
0           0  Jonathan Taylor*+  IND  RB   22  17  17   51.0   40.0   
1           1      Cooper Kupp*+  LAR  WR   28  17  17  191.0  145.0   
2           2     Deebo Samuel*+  SFO  WR   25  16  15  121.0   77.0   
3           3         Josh Allen  BUF  QB   25  17  17    0.0    0.0   
4           4      Austin Ekeler  LAC  RB   26  16  16   94.0   70.0   

   PassingYds  ...  PassingAtt  RushingYds  RushingTD  RushingAtt  \
0         0.0  ...         0.0      1811.0       18.0       332.0   
1         0.0  ...         1.0        18.0        0.0         4.0   
2        24.0  ...         2.0       365.0        8.0        59.0   
3      4407.0  ...       646.0       763.0        6.0       122.0   
4         0.0  ...         0.0       911.0       12.0       206.0   

   ReceivingYds  ReceivingTD  FantasyPoints   Int  Fumbles  FumblesLost  
0         360.0          2.0          333.0   0.0      4.0          2.0  
1        1947.0         16.0          295.0   0.0      0.0          0.0  
2        1405.0          6.0          262.0   0.0      4.0          2.0  
3           0.0          0.0          403.0  15.0      8.0          3.0  
4         647.0          8.0          274.0   0.0      4.0          3.0  

[5 rows x 21 columns]

In [2]:
import tensorflow as tf


print(tf.__version__)
print(tf.config.list_physical_devices('GPU'))

2.12.0
[PhysicalDevice(name='/physical_device:GPU:0', device_type='GPU')]


In [3]:
df.dtypes

Unnamed: 0         int64
Player            object
Tm                object
Pos               object
Age                int64
G                  int64
GS                 int64
Tgt              float64
Rec              float64
PassingYds       float64
PassingTD        float64
PassingAtt       float64
RushingYds       float64
RushingTD        float64
RushingAtt       float64
ReceivingYds     float64
ReceivingTD      float64
FantasyPoints    float64
Int              float64
Fumbles          float64
FumblesLost      float64
dtype: object

In [ ]:
df.isna().sum()


Unnamed: 0        0
Player            0
Tm                0
Pos              75
Age               0
G                 0
GS                0
Tgt               1
Rec               1
PassingYds        1
PassingTD         1
PassingAtt        1
RushingYds        1
RushingTD         1
RushingAtt        1
ReceivingYds      1
ReceivingTD       1
FantasyPoints    90
Int               1
Fumbles           1
FumblesLost       1
dtype: int64

In [ ]:
df.isnull().sum().sum()
#total of 178 null values

178

In [ ]:
df.nunique()


Unnamed: 0       671
Player           671
Tm                34
Pos                4
Age               20
G                 18
GS                18
Tgt              117
Rec               94
PassingYds        83
PassingTD         31
PassingAtt        69
RushingYds       195
RushingTD         15
RushingAtt       113
ReceivingYds     314
ReceivingTD       16
FantasyPoints    180
Int               17
Fumbles           15
FumblesLost        7
dtype: int64

In [ ]:
pts_mean = df['FantasyPoints'].mean()
pts_mean
#Each player has an avergae of approximately 61 fantasy points.


60.602409638554214

In [ ]:
#Average Player Age
age_mean = df ['Age'].mean()
age_mean
#The average NFL Fantasy Football player was approximately 26 years old in the year 2021.

26.2354694485842

In [ ]:
#Average Passing Yards
pass_mean = df['PassingYds'].mean()
pass_mean
#The average passing yards amongst all players is 198 yds.

198.0820895522388

In [ ]:
df.describe()

Unnamed: 0         Age           G          GS         Tgt         Rec  \
count  671.000000  671.000000  671.000000  671.000000  670.000000  670.000000   
mean   335.000000   26.235469   10.690015    4.842027   26.926866   18.073134   
std    193.845299    3.250900    5.501902    5.601873   36.167758   24.388283   
min      0.000000   21.000000    0.000000    0.000000    0.000000    0.000000   
25%    167.500000   24.000000    6.000000    0.000000    1.000000    0.000000   
50%    335.000000   26.000000   12.000000    2.000000   10.000000    6.000000   
75%    502.500000   28.000000   16.000000    9.000000   41.750000   28.000000   
max    670.000000   44.000000   17.000000   17.000000  191.000000  145.000000   

        PassingYds   PassingTD  PassingAtt   RushingYds   RushingTD  \
count   670.000000  670.000000  670.000000   670.000000  670.000000   
mean    198.082090    1.253731   27.910448    93.477612    0.753731   
std     797.326876    5.435155  110.096633   222.047724    2.017993   
min       0.000000    0.000000    0.000000   -13.000000    0.000000   
25%       0.000000    0.000000    0.000000     0.000000    0.000000   
50%       0.000000    0.000000    0.000000     0.000000    0.000000   
75%       0.000000    0.000000    0.000000    46.750000    0.000000   
max    5316.000000   43.000000  719.000000  1811.000000   18.000000   

       RushingAtt  ReceivingYds  ReceivingTD  FantasyPoints         Int  \
count  670.000000    670.000000   670.000000     581.000000  670.000000   
mean    21.613433    198.122388     1.243284      60.602410    0.656716   
std     50.036562    296.971711     2.237520      72.989054    2.566615   
min      0.000000     -4.000000     0.000000      -4.000000    0.000000   
25%      0.000000      0.000000     0.000000       8.000000    0.000000   
50%      1.000000     52.000000     0.000000      35.000000    0.000000   
75%     10.000000    276.500000     2.000000      86.000000    0.000000   
max    332.000000   1947.000000    16.000000     403.000000   17.000000   

          Fumbles  FumblesLost  
count  670.000000   670.000000  
mean     0.885075     0.379104  
std      1.940116     0.874577  
min      0.000000     0.000000  
25%      0.000000     0.000000  
50%      0.000000     0.000000  
75%      1.000000     0.000000  
max     14.000000     6.000000

In [ ]:
player_passing_yards = df.groupby('Player')['PassingYds'].sum()

# Find the player with the most passing yards
player_with_most_passing_yards = player_passing_yards.idxmax()
most_passing_yards = player_passing_yards.max()
print(f"The player with the most passing yards is {player_with_most_passing_yards} with {most_passing_yards} yards.")


The player with the most passing yards is Tom Brady* with 5316.0 yards.


In [ ]:
#Find the player with the least passing yads
player_passing_yards = df.groupby('Player')['PassingYds'].sum()


player_with_least_passing_yards = player_passing_yards.idxmin()
least_passing_yards = player_passing_yards.min()

print(f"The player with the least passing yards is {player_with_least_passing_yards} with {least_passing_yards} yards.")

The player with the least passing yards is A.J. Brown with 0.0 yards.


In [ ]:
#Find the player with the most rushing yards
player_rushing_yards = df.groupby('Player')['RushingYds'].sum()

player_with_most_rushingyds = player_rushing_yards.idxmax()
most_rushingyds = player_rushing_yards.max()
print(f"The player with the most rushing yards is {player_with_most_rushingyds} with {most_rushingyds}.")


The player with the most rushing yards is Jonathan Taylor*+ with 1811.0.


In [ ]:
#Player with the least rushing yards
player_rushing_yards = df.groupby('Player')['RushingYds'].sum()

player_with_least_rushingyds = player_rushing_yards.idxmin()
least_rushingyds = player_rushing_yards.min()
print(f"The player with the least rushing yards is {player_with_least_rushingyds} with {least_rushingyds}.")

The player with the least rushing yards is Darius Slayton with -13.0.


In [ ]:
#Player with the most rushing touch downs
player_rushing_td = df.groupby('Player')['RushingTD'].sum()
player_with_most_rushingtd = player_rushing_td.idxmax()
most_rushingtd = player_rushing_td.max()
print(f"The player with the most rushing touchdowns is {player_with_most_rushingtd} with {most_rushingtd}.")




The player with the most rushing touchdowns is Jonathan Taylor*+ with 18.0.


In [ ]:
#Player with the most receiving touchdowns
player_receiving_td = df.groupby('Player')['ReceivingTD'].sum()
player_with_most_receivingtd = player_receiving_td.idxmax()
most_receivingtd = player_receiving_td.max()
print(f"The player with the most receiving touchdowns is {player_with_most_receivingtd} with {most_receivingtd}.")



The player with the most receiving touchdowns is Cooper Kupp*+ with 16.0.


In [ ]:
#Player with most receiving yards

player_receiving_yd = df.groupby('Player')['ReceivingYds'].sum()
player_with_most_receivingyd = player_receiving_yd.idxmax()
most_receivingyd = player_receiving_yd.max()
print(f"The player with the most receiving yards is {player_with_most_receivingyd} with {most_receivingyd}.")

The player with the most receiving yards is Cooper Kupp*+ with 1947.0.


In [ ]:
#Player with msot fumbles

player_fumbles = df.groupby ('Player')['Fumbles'].sum()
player_with_most_fumbles= player_fumbles.idxmax()
most_fumbles = player_fumbles.max()
print(f"The player with the most fumbles is {player_with_most_fumbles} with {most_fumbles}.")

The player with the most fumbles is Dak Prescott with 14.0.


In [5]:
#Player with the least fumbles

player_fumbles = df.groupby ('Player')['Fumbles'].sum()
player_with_least_fumbles= player_fumbles.idxmin()
least_fumbles = player_fumbles.min()
print(f"The player with the least fumbles is {player_with_least_fumbles} with {least_fumbles}.")

The player with the least fumbles is A.J. Brown with 0.0.


In [9]:
#Player with most fantasy points

player_fantasypts = df.groupby ('Player')['FantasyPoints'].sum()
player_with_most_fantasypts = player_fantasypts.idxmax()
most_fantasypts = player_fantasypts.max()
print(f"The player with the most fantasy points is {player_with_most_fantasypts} with a total of {most_fantasypts}.")


The player with the most fantasy points is Josh Allen with a total of 403.0.


In [11]:
#Player with the least fantasy points

player_fantasypts = df.groupby ('Player')['FantasyPoints'].sum()
player_with_least_fantasypts = player_fantasypts.idxmin()
least_fantasypts = player_fantasypts.min()
print(f"The player with the least fantasy points is {player_with_least_fantasypts} with a total of {least_fantasypts}.")

The player with the least fantasy points is Diontae Spencer with a total of -4.0.


In [21]:
#printing a table
data = {
    'Player': ['Tom Brady', 'Johnathon Taylor', 'Johnathon Taylor', 'Cooper Kupp', 'Cooper Kupp','Dak Prescott', 'Josh Allen'],

    'Player Maximum': ['Most Passing Yards: 5316.0 yds', 'Most Rushing Yards: 1811.0 yds', 'Most Rushing Touchdowns: 18.0', 'Most Receiving Touchdowns: 16.0', 'Most Receiving Yards: 1947.0 yds', 'Most Fumbles: 14.0', 'Most Fantasy Points: 403.0 pts']
}


df = pd.DataFrame(data)


print(df)

             Player                    Player Maximum
0         Tom Brady    Most Passing Yards: 5316.0 yds
1  Johnathon Taylor    Most Rushing Yards: 1811.0 yds
2  Johnathon Taylor     Most Rushing Touchdowns: 18.0
3       Cooper Kupp   Most Receiving Touchdowns: 16.0
4       Cooper Kupp  Most Receiving Yards: 1947.0 yds
5      Dak Prescott                Most Fumbles: 14.0
6        Josh Allen    Most Fantasy Points: 403.0 pts



 All in all, according to the data Tom Brady has the most amount of passing yards with a total of 5316.0 yards. On the other hand, A.J. Brown has the least amount of passing yards with a total of 0.0 yards, and also has the least amount of fumbles with a total of 0.0 fumbles. The player with the most rushing yards and rushing touchdowns is Johnathon Taylor with a total of 1811 rushing yards and 18.0 rushing touchdowns. The player with the most receiving yards and receiving touchdowns is Cooper Kupp with a total of 1947.0 receiving yards and 16.0 receiving touchdowns. Dak Prescott is found to have the most amount of fumbles with 14 fumbles. Finally, Josh Allen has the most amount of fantasy points with a total of 403.0 points.

















